src/components/ftbot/PairlistConfigurator.vue

```diff
@@ -0,0 +1,272 @@
+  'error',
+]);
+const props = defineProps<{
+  selectedConfig: PairlistConfig;
+}>();
+
+const botStore = useBotStore();
+
+const availablePairlists = ref<Pairlist[]>([]);
+const config = ref<PairlistConfig>(props.selectedConfig);
+const pairlistConfigsEl = ref<HTMLElement | null>(null);
+const availablePairlistsEl = ref<HTMLElement | null>(null);
+const evaluating = ref(false);
+
+const firstPairlistIsGenerator = computed<boolean>(() => {
+  // First pairlist must be a generator
+  if (config.value.pairlists[0]?.is_pairlist_generator) {
+    return true;
+  }
+  return false;
+});
+
+const pairlistValid = computed<boolean>(() => {
+  return firstPairlistIsGenerator.value && config.value.pairlists.length > 0;
+});
+
+// v-for updates with sorting, deleting and adding items seem to get wonky without unique keys for every item
+const pairlistsComp = computed(() =>
+  config.value.pairlists.map((p) => {
+    if (p.id) {
+      return p;
+    } else {
+      return { id: Date.now().toString(36) + Math.random().toString(36).substring(2), ...p };
+    }
+  }),
+);
+
+const configJSON = computed(() => {
+  return JSON.stringify(configToPayloadItems(), null, 2);
+});
+
+useSortable(availablePairlistsEl, availablePairlists.value, {
+  group: {
+    name: 'configurator',
+    pull: 'clone',
+    put: false,
+  },
+  sort: false,
+  filter: '.no-drag',
+});
+
+useSortable(pairlistConfigsEl, config.value.pairlists, {
+  handle: '.handle',
+  group: 'configurator',
+  onUpdate: async (e) => {
+    moveArrayElement(config.value.pairlists, e.oldIndex, e.newIndex);
+  },
+  onAdd: (e) => {
+    const pairlist = availablePairlists.value[e.oldIndex];
+    addToConfig(pairlist, e.newIndex);
+    // quick fix from: https://github.com/SortableJS/Sortable/issues/1515
+    e.clone.replaceWith(e.item);
+    e.clone.remove();
+  },
+});
+
+onMounted(async () => {
+  availablePairlists.value = (await botStore.activeBot.getPairlists()).pairlists.sort((a, b) =>
+    // Sort by is_pairlist_generator (by name), then by name.
+    // TODO: this might need to be improved
+    a.is_pairlist_generator === b.is_pairlist_generator
+      ? a.name.localeCompare(b.name)
+      : a.is_pairlist_generator
+      ? -1
+      : 1,
+  );
+});
+
+function addToConfig(pairlist: Pairlist, index: number) {
+  pairlist = structuredClone(toRaw(pairlist));
+  for (const param in pairlist.params) {
+    pairlist.params[param].value = pairlist.params[param].default
+      ? pairlist.params[param].default.toString()
+      : '';
+  }
+  config.value.pairlists.splice(index, 0, pairlist);
+}
+
+function removeFromConfig(index: number) {
+  config.value.pairlists.splice(index, 1);
+}
+
+async function save() {
+  emit('saveConfig', config.value);
+}
+
+async function evaluateClick() {
+  const payload = configToPayload();
+
+  evaluating.value = true;
+  const res = await botStore.activeBot.evaluatePairlist(payload);
+  emit('started', res);
+  const evalIntervalId = setInterval(async () => {
```

Comment:
Would we want to clear the interval also if the user leaves the page? (basically on unmounted) - or will that clean itself up automatically?

Reply:
Ah good catch, I think we do and it doesn't. Also, is it a concern if the user starts evaluation, leaves the page and starts a new one before the first finished? Maybe should extract the logic to a store?

Reply:
I've been thinking about that part - i think i'll want to do this differently (from a backend perspective) - mainly with the goal to be able to reuse most of the logic elsewhere, not necessarily to benefit this particular endpoint.

the way i thought about it:
* the post request returns an ID
* query a generic "/bgjob/<id>" endpoint - which provides the status
* once the status changes to success (or whatever) - query the original endpoint (with the ID attached) to get the actual result

while it's effectively one more call - we can probably make that reusable in a "ping until ready" composable - and make this same logic reusable for multiple background tasks.

Opinions (before i start implementing this) ... ?

Reply:
yeah, i think so, i'll try to have a look at this soon